accumulator_updater/programs/accumulator_updater/src/instructions/emit_inputs.rs

```diff
@@ -0,0 +1,165 @@
+use {
+    crate::{
+        accumulator_acc_seeds,
+        accumulator_acc_seeds_with_bump,
+        state::*,
+        AccumulatorUpdaterError,
+    },
+    anchor_lang::{
+        prelude::*,
+        system_program::{
+            self,
+            CreateAccount,
+        },
+    },
+};
+
+
+pub fn emit_inputs<'info>(
+    ctx: Context<'_, '_, '_, 'info, EmitInputs<'info>>,
+    base_account: Pubkey,
+    data: Vec<Vec<u8>>,
+    account_type: u32,
+    account_schemas: Vec<u8>,
```

Comment:
I think it would be nice if the interface of this function resembled a commonly-used data structure so that callers had a natural reference point for what to expect. I think the accumulator is most similar to a `Map` where the keys are `(base_account, account_schema)` and the values are arbitrary data. Given that, I would try to make the arguments map on to the function signature for inserting multiple items in a Map. E.g., I think a function signature like

```
/// Insert elements into the Map identified by the calling program and map_id.
/// Each element is written into a separate account (explain how to derive the PDA). 
/// The caller of this instruction must pass those PDAs while calling this function in the same order as elements. 
put_all(ctx, map_id: Pubkey, elements: <Vec<(u8, Vec<u8>)>>)
```

would make more sense. (map_id is base_account)